### ISSUE TITLE
GetAxis for move input

### ISSUE THREAD

**Issue by delgelato** [vis2k/uMMORPG_CE#13]
There's a few parts to the problem:
1. Input lingers due to using GetAxis + normalized
2. In the current setup, deceleration is actually never used (it always completely stops, even if deceleration is 0) because moveDir is multiplied with desiredDir which is 0 when no input, and this overrides deceleration process

#1 In PlayerCharacterControllerMovement.285
```
horizontal = Input.GetAxis("Horizontal");
vertical = Input.GetAxis("Vertical");
...
return new Vector2(horizontal, vertical).normalized;
```
Those combined, makes input linger. While axis nears 0, it'll always be normalized
Would GetAxisRaw work better?

#2 Acceleration works as expected (setting it to 1 makes it accelerate really slow). But Deceleration doesn't work. horizontalSpeed is supposed to decelerate in RUNNING state, but regardless of the #1 fix is applied, it will jump to 0 because in .1017
`Vector3 desiredDir = GetDesiredDirection(inputDir);`
At this point, inputDir will be 0. Which means
`moveDir.x = desiredDir.x * horizontalSpeed;`
Will also instantly be 0

So i changed GetDesiredDirection to:
```
if(inputDir == Vector2.zero)
{
     return moveDir.normalized;            
}else
     return transform.forward * inputDir.y + transform.right * inputDir.x;
```
Because when there's no input,the only reference to decelerate, is the current remaining moveDir, not inputDir which will certainly be 0
Deceleration now works properly

**Comment by miwarnec** [comment]
thanks for sharing, will take a look

**Comment by miwarnec** [comment]
Checking.. this is what happens with normalized input direction right now:
![image](https://user-images.githubusercontent.com/16416509/82425908-74db1a00-9a87-11ea-9c00-00eb9121fb6f.png)

so it's basically GetAxisRaw unintentionally. will check if that's good or not.

**Comment by delgelato** [comment]
The only problem is, deceleration never gets called. Set it to super low or even 0, and speed will NOT slow down.
GetAxis will gradually turn to 0, but before it's completely 0, speed will be still be 1 (bcoz it's normalized).
When it finally reaches 0, speed becomes 0 bcoz of it being multiplied by it. So there's nothing in between for deceleration to apply to horizontalSpeed

**Comment by miwarnec** [comment]
@delgelato fix is in V2.15.
GetInput now only normalizes if length > 1, so that GetAxis acceleration/deceleration is used.
The custom acceleration/deceleration based on state was fully removed everywhere.
It's a lot more simple and works as expected now.
Nice find!

**Comment by miwarnec** [comment]
Note that Input Settings -> Horizontal/Vertical Axis -> Gravity can be modified to have faster/slower acceleration now, depending on input device etc.
